include/ensmallen_bits/eve/eve.hpp

```diff
@@ -106,8 +106,8 @@ class Eve
            typename MatType,
            typename GradType,
            typename... CallbackTypes>
-  typename std::enable_if<IsArmaType<GradType>::value,
-      typename MatType::elem_type>::type
+  typename std::enable_if<IsArmaType<GradType>::value ||
+      coot::is_coot_type<GradType>::value, typename MatType::elem_type>::type
```

Comment:
One alternate idea here would be to make a combined `IsArmaOrCootType` traits class and use that instead.